### PR TITLE
Update ApplyImpulseResponse to support multichannel IRs and convolution

### PR DESCRIPTION
**Pull request by crlandsc**
I discovered that all IRs were only being loaded in as mono IRs. I updated the transform so that it can support both mono and multichannel IRs and channel-wise convolution to retain the correct spatial information.

### CODE REVIEW

File: audiomentations/augmentations/apply_impulse_response.py [hunk 1]
```diff
@@ -68,14 +70,18 @@ def apply(self, samples: NDArray[np.float32], sample_rate: int):
...
-        else:
-            signal_ir = convolve(samples, ir)
+        # Expand dimensions to match
+        if samples.ndim == 1:
+            samples = samples[None, :]
+        if ir.ndim == 1:
+            ir = ir[None, :]
+
+        # Loop over all samples channels for channelwise convolution
+        signal_ir = []
```
[the remark this answers is not in the document]
Comment (crlandsc) [reply audiomentations/augmentations/apply_impulse_response.py]: This was a good suggestion, I just adapted the code to incorporate this approach. Thanks!

**Comment by crlandsc** [comment]
I updated the code with your suggestion for memory efficiency and cleaned it up a little more. I also added a stereo IR to verify compatibility with the unittest (which checks for both mono and stereo convolution). Everything checked out on my end, but let me know if there are other updates that should be made.

**Comment by iver56** [comment]
Thanks! I will take a look at it tomorrow

**Comment by crlandsc** [comment]
Just merged your PR!

**Review by iver56 — APPROVED** [review]
Thanks for the contribution! This LGTM. I see that I also need to update test_find_audio_files_multiple_dirs, but I can do that after merge

**Comment by crlandsc** [comment]
Great thanks so much!

**Pull request merged**